x/twap/logic_test.go

```diff
@@ -31,18 +31,55 @@ func newRecord(poolId uint64, t time.Time, sp0, accum0, accum1 sdk.Dec) types.Tw
 		// make new copies
 		P0ArithmeticTwapAccumulator: accum0.Add(sdk.ZeroDec()),
 		P1ArithmeticTwapAccumulator: accum1.Add(sdk.ZeroDec()),
-	}
+	}}
 }
 
 // make an expected record for math tests, we adjust other values in the test runner.
-func newExpRecord(accum0, accum1 sdk.Dec) types.TwapRecord {
-	return types.TwapRecord{
+func newExpRecord(poolId uint64, accum0, accum1 sdk.Dec) []types.TwapRecord {
+	return []types.TwapRecord{{
+		PoolId:      poolId,
 		Asset0Denom: defaultTwoAssetCoins[0].Denom,
 		Asset1Denom: defaultTwoAssetCoins[1].Denom,
 		// make new copies
 		P0ArithmeticTwapAccumulator: accum0.Add(sdk.ZeroDec()),
 		P1ArithmeticTwapAccumulator: accum1.Add(sdk.ZeroDec()),
+	}}
+}
+
+func newTapRecord(poolId uint64, t time.Time, sp0, accum0, accum1 sdk.Dec) []types.TwapRecord {
+	twapAB := types.TwapRecord{
+		PoolId:          poolId,
+		Asset0Denom:     defaultThreeAssetCoins[0].Denom,
+		Asset1Denom:     defaultThreeAssetCoins[1].Denom,
+		Time:            t,
+		P0LastSpotPrice: sp0,
+		P1LastSpotPrice: sdk.OneDec().Quo(sp0),
+		// make new copies
+		P0ArithmeticTwapAccumulator: accum0.Add(sdk.ZeroDec()),
+		P1ArithmeticTwapAccumulator: accum1.Add(sdk.ZeroDec()),
 	}
+	twapAC := twapAB
+	twapAC.Asset1Denom = denom2
+	twapBC := twapAC
+	twapBC.Asset0Denom = denom1
+	return []types.TwapRecord{twapAB, twapAC, twapBC}
+}
+
+// make an expected record for math tests, we adjust other values in the test runner.
+func newTapExpRecord(poolId uint64, accum0, accum1 sdk.Dec) []types.TwapRecord {
```

Comment:
Is there a reason we cannot use newTapRecord? I feel like if we are to adjust other values in the test runner any ways, we can re use `newTapRecord`